packages/directlinespeech/src/createAdapters.js

```diff
@@ -25,25 +27,28 @@ export default async function create({
   speechSynthesisOutputFormat,
   textNormalization,
   userID,
-  username
+  username,
+  enableInternalHttpSupport = false
 }) {
   if (!fetchCredentials) {
     throw new Error('"fetchCredentials" must be specified.');
   }
 
-  const { authorizationToken, region, subscriptionKey } = await resolveFunctionOrReturnValue(fetchCredentials);
+  const { authorizationToken, region, subscriptionKey, directLineToken } = await resolveFunctionOrReturnValue(fetchCredentials);
 
   if (
     (!authorizationToken && !subscriptionKey) ||
     (authorizationToken && subscriptionKey) ||
     (authorizationToken && typeof authorizationToken !== 'string') ||
-    (subscriptionKey && typeof subscriptionKey !== 'string')
+    (subscriptionKey && typeof subscriptionKey !== 'string') ||
+    (enableInternalHttpSupport && !directLineToken)
   ) {
     throw new Error(
-      '"fetchCredentials" must return either "authorizationToken" or "subscriptionKey" as a non-empty string only.'
+      '"fetchCredentials" must return either "authorizationToken" or "subscriptionKey" as a non-empty string only. If enableInternalHttpSupport = true, then it should also return a directLineToken'
```

Comment:
Can you make this sentence better?

Reply:
took a stab and changed to 

``` 
throw new Error(
      '"fetchCredentials" must return either "authorizationToken" or "subscriptionKey" as a non-empty string only. If enableInternalHTTPSupport is set to true, then it should also return a non-empty "directLineToken"'
    );
```

suggestions?